docs/backends/spdk/index.rst

```diff
@@ -0,0 +1,365 @@
+longer know that the PCIe device is an NVMe device, thus, it no longer lives in
+Linux devfs, that is, no longer available in ``/dev`` as e.g. ``/dev/nvme0n1``.
+
+Instead of the filepath in devfs, then you use PCI ids and xNVMe options.
+
+As always, use the ``xnvme`` cli tool to enumerate devices:
+
+.. literalinclude:: 400_xnvme_enum.cmd
+   :language: bash
+
+.. literalinclude:: 400_xnvme_enum.out
+   :language: bash
+
+Notice that multiple URIs using the same PCI id but with different **xNVMe**
+``?opts=<val>``. This is provided as a means to tell **xNVMe** that you want to
+use the NVMe controller at ``0000:03:00.0`` and the namespace identified by
+``nsid=1``.
+
+.. literalinclude:: 410_xnvme_info.cmd
+   :language: bash
+
+.. literalinclude:: 410_xnvme_info.out
+   :language: bash
+
+
+Similarly, when using the API, then you would use these URIs instead of
+filepaths::
+
+  ...
+  struct xnvme_dev *dev = xnvme_dev_open("pci:0000:01:00.0?nsid=1");
+  ...
+
+
+No devices found
+~~~~~~~~~~~~~~~~
+
+When running ``xnvme enum`` and the output-listing is empty, then there are no
+devices. When running with ``vfio-pci``, this can occur when your devices are
+sharing iommu-group with other devices which are still bound to in-kernel
+drivers. This could be NICs, GPUs or other kinds of peripherals.
+
+The division of devices into groups is not something that can be easily
+switched, but you can try to manually unbind the other devices in the iommu
+group from their kernel drivers.
+
+If that is not an option then you can try to re-organize your physical
+connectivity of deviecs, e.g. move devices around.
+
+Lastly you can try using ``uio_pci_generic`` instead, this can most easily be
+done by disabling iommu by adding the kernel option: ``iommu=off`` to the
+kernel command-line and rebooting.
+
+.. _sec-backends-spdk-instrumentation:
+
+Instrumentation
+---------------
+
+...
+
+.. _sec-backends-spdk-config:
+
+System Configuration
+--------------------
+
+...
+
+Driver Attachment and memory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, then it is the operating system responsibility to provide device
+drivers, thus by default, then the operating system has bound its NVMe driver
+to the NVMe devices attached to your system. For a user-space driver to operate,
+then the operating system driver must be detached and bound to a driver such as
+``vfio-pci`` or ``uio-generic-pci``, such that the NVMe driver can operate in
+user-space.
+
+By running the command below **8GB** of hugepages will be configured and the
+device detached from the Kernel NVMe driver:
+
+.. literalinclude:: 010_xnvme_driver.cmd
+   :language: bash
+
+The ``xnvme-driver`` script is a merge of the **SPDK** ``setup.sh`` script and
+its dependencies.
+
+The command above should produce output similar to:
+
+.. literalinclude:: 010_xnvme_driver.out
+   :language: bash
+
+If anything else that the above is output from ``setup.sh``, for example::
+
```

Comment:
We probably don't mean "setup.sh" here

Reply:
Please skip the backends and tools sections, as they are incomplete and may not make sense. These sections were not addressed due to the substantial size of this PR.